x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java

```diff
@@ -147,36 +152,49 @@ private Object extractNestedField(SearchHit hit) {
         return value;
     }
 
-    protected Object unwrapFieldsMultiValue(Object values) {
+    protected Object unwrapFieldsMultiValue(Object values, boolean allowMultivalue) {
```

Comment:
Isn't allowMultivalue better as a class parameter vs a method one? Are there cases where the same instance flips between allowing and not allowing multi values?

Reply:
I agree with you, I don't see a reason why the behavior should flop between allowing and not allowing multi-values.
Basically, this component is used by SQL (that does not allow multi-values) and EQL (that allows them), so it can be set when the instance is created. 
This change will also reduce the impact on the APIs, I can completely remove `extract(SearchHit hit, boolean allowMultivalue)`